source/time_tables/thermometer.cpp

```diff
@@ -130,12 +132,72 @@ void Thermometer::generate_thermometer(const type::Route* route) {
     generate_thermometer(std::vector<vector_idx>(stop_point_lists.begin(), stop_point_lists.end()));
 }
 
+// Define types for next function 'generate_topological_thermometer'
+struct VertexProperties
+{
+    type::idx_t idx;
+    VertexProperties() : idx(0) {}
+    VertexProperties(type::idx_t const& i): idx(i) {}
+};
+typedef boost::adjacency_list< boost::vecS, boost::vecS, boost::directedS, boost::property<boost::vertex_bundle_t, VertexProperties> > Graph;
+typedef boost::graph_traits<Graph>::vertex_descriptor Vertex;
+typedef std::vector< Vertex > Container;
+
+// Generate an topologicaly exact thermometer
+// return false if topological sort fail (because of a cycle)
+// return true if succeed
+bool Thermometer::generate_topological_thermometer(const std::vector<vector_idx> &stop_point_lists) {
+
+    Graph stop_point_graph;
+    std::map<type::idx_t, Vertex> vertex_map;
+    std::map<type::idx_t, Vertex>::iterator it;
+    for(auto v : stop_point_lists) {
```

Comment:
for consistency, we put a space between `for` and `(`. Same for `if`, `switch`, `while` and such keywords.